go.mod

```diff
@@ -221,9 +221,29 @@ replace (
 
 // HV2 related packages
 replace (
-	cosmossdk.io/simapp => ./simapp
+	// same versions tagged on our fork
+	cosmossdk.io/api => github.com/0xPolygon/cosmos-sdk/api v0.7.2
+	cosmossdk.io/client/v2 => github.com/0xPolygon/cosmos-sdk/client/v2 v2.0.0-beta.1
+	cosmossdk.io/collections => github.com/0xPolygon/cosmos-sdk/collections v0.4.0
+	cosmossdk.io/core => github.com/0xPolygon/cosmos-sdk/core v0.11.0
+	cosmossdk.io/depinject => github.com/0xPolygon/cosmos-sdk/depinject v1.0.0-alpha.4
+	cosmossdk.io/errors => github.com/0xPolygon/cosmos-sdk/errors v1.0.0
+	cosmossdk.io/log => github.com/0xPolygon/cosmos-sdk/log v1.2.1
+	cosmossdk.io/math => github.com/0xPolygon/cosmos-sdk/math v1.2.0
+	cosmossdk.io/simapp => github.com/0xPolygon/cosmos-sdk/simapp v0.0.0-20240409091007-84d9b34a8681
```

Comment:
For this case (when you have modifications on the module itself so you have to produce a hash-based pseudo-version) it works as well, but you will need to remember to update it here every time you make changes to `simapp`.

To avoid this (cause it's tedious) you might rather want to try using the relative path (`./simapp`). With this, you don't need to keep on changing the version here all the time. However, with this, you might not be able to do `go run github.com/0xPolygon/cosmos-sdk...` or `go install ...`. If that's not part of your workflow, then the relative path may be best while you are in development (and only set an official version here once you release the Heimdall's SDK fork)

Reply:
Wrt relative path import, I started with it actually, then changed it back because I am not a big fan of it :D
However, I must admit it helps in such long living pre-releases development phase. I guess I'll change it, thanks

Reply:
Done https://github.com/0xPolygon/cosmos-sdk/pull/16/commits/1554abab5ff582a27ef21c3d61f33e8494f98801